app/models/opds/publication.rb

```diff
@@ -0,0 +1,248 @@
+        # when :artist
+        # when :colorist
+        # when :illustrator
+        # when :inker
+        # when :letterer
+        # when :narrator
+        # when :penciler
+        # when :translator
+        # when :other
+        when :author
+          @monograph.contributors.each do |contributor|
+            rvalue.append(contributor) unless /\(\s*\S+\s*\)/i.match?(contributor)
+          end
+        when :editor
+          @monograph.contributors.each do |contributor|
+            m = /(\S+)(\s*\(\s*editor\s*\).*)/i.match(contributor)
+            rvalue.append(m[1]) if m.present?
+          end
+        end
+
+        return nil if rvalue.blank?
+        return rvalue.first if rvalue.count == 1
+        rvalue
+      end
+
+      def description
+        # plain text
+        # description_tesim
+        return nil if @monograph.description.blank?
+        MarkdownService.markdown_as_text(@monograph.description, true)
+      end
+
+      def duration
+        # seconds
+        nil
+      end
+
+      def identifier
+        # doi_sim
+        # hdl_sim
+        # identifier_tesim
+        @monograph.identifier # "https://uri"
+      end
+
+      def imprint
+      end
+
+      def language
+        # BCP 47 e.g. 'en'
+        # language_tesim
+        return nil if @monograph.language.blank?
+        return 'en' if /\s*(en)(glish)?\s*/i.match?(@monograph.language)
+        nil
+      end
+
+      def modified
+        # ISO 8601
+        # timestamp
+        # system_create_dtsi
+        # system_modified_dtsi
+        # date_uploaded_dtsi
+        # date_modified_dtsi
+      end
+
+      def numberOfPages
+      end
+
+      def published
+        # ISO 8601
+        # date_created_tesim
+        @monograph.published&.iso8601
+      end
+
+      def publisher
+        # publisher_tesim
+        @monograph.publisher
+      end
+
+      def readingProgression
+        # 'ltr', 'rtl', 'ttb', 'btt', or 'auto' (default)
+      end
+
+      def series
+        # optional links
+        # see also belongsTo and collection
+        # series_tesim
+        @monograph.series
+      end
+
+      def subject
+        # subject_tesim
+        return nil if @monograph.subjects.blank?
+        return @monograph.subjects.first if @monograph.subjects.count == 1
+        @monograph.subjects
+      end
+
+      def subtitle
+      end
+
+      def title
+        # title_tesim
+        # {
+        #     en: @monograph.title
+        # }
+        # MarkdownService.markdown_as_text(@monograph.title, true)
+        @monograph.title
+      end
+
+      def title_sort_as
+      end
+
+      def download_ebook_url(ebook)
+        Rails.application.routes.url_helpers.download_ebook_url(ebook.noid)
+      end
+
+      def monograph_image_url(monograph, width_size = nil)
+        width_size ||= 'full'
+        Riiif::Engine.routes.url_helpers.image_url(monograph.noid, host: Rails.application.routes.url_helpers.root_url, size: width_size, format: 'jpg')
```

Comment:
We'll need to use the cover (file_set) noid here instead of the monographs noid (see note in HELIO-3483)

Reply:
Switched to cover for images.